packages/web-extension/test/walletManager/WalletRepository.test.ts

```diff
@@ -206,6 +209,45 @@ describe('WalletRepository', () => {
       ]);
     });
 
+    it('allows creating 1852 and 1854 purpose account with the same index', async () => {
+      const standardAccountProps = createAccount(
+        0,
+        storedLedgerWallet.accounts[storedLedgerWallet.accounts.length - 1].accountIndex + 1,
+        KeyPurpose.STANDARD
+      );
+      const standardProps = {
+        ...standardAccountProps,
+        walletId: storedLedgerWallet.walletId
+      };
+
+      const multiSigAccountProps2 = createAccount(
+        0,
+        storedLedgerWallet.accounts[storedLedgerWallet.accounts.length - 1].accountIndex + 1,
+        KeyPurpose.MULTI_SIG
+      );
+
+      const multiSigProps = {
+        ...multiSigAccountProps2,
+        walletId: storedLedgerWallet.walletId
+      };
+
+      await expect(repository.addAccount(standardProps)).resolves.toEqual(standardProps);
+      expect(store.setAll).toBeCalledWith([
+        {
+          ...storedLedgerWallet,
+          accounts: [...storedLedgerWallet.accounts, standardAccountProps]
+        }
+      ]);
+
+      await expect(repository.addAccount(multiSigProps)).resolves.toEqual(multiSigProps);
+      expect(store.setAll).toBeCalledWith([
+        {
+          ...storedLedgerWallet,
+          accounts: [...storedLedgerWallet.accounts, multiSigAccountProps2]
```

Comment:
Isn't this a bug? Seems like the previously added `standardAccountProps` is gone.
```suggestion
          accounts: [...storedLedgerWallet.accounts, standardAccountProps, multiSigAccountProps2]
```

Reply:
I looked into why this happens briefly, but will take a closer look.